components/Select.jsx

```diff
@@ -1,20 +1,24 @@
 import styles from 'styles/Select.module.css'
 import { useTranslate } from 'hooks/useTranslate'
+import { useLocale } from 'hooks/useLocale'
+
+const REPORT_DATE_REGEXP = /(?<year>[0-9]{4})(?<month>[0-9]{2})(?<day>[0-9]{2})/
+
+const toDate = (value) => {
+  const { year, month, day } = value.match(REPORT_DATE_REGEXP).groups
+  return new Date(parseInt(year), parseInt(month) - 1, parseInt(day))
+}
+
+const formatDate = ({ locale, value }) => {
+  const date = toDate(value)
+  return new Intl.DateTimeFormat(locale, {
+    year: 'numeric', month: '2-digit', day: '2-digit'
+  }).format(date)
+}
```

Comment:
Creo que podría ser mejor sacar las opciones en una constante fuera de la llamada del método.

```suggestion
  const options = {
    year: 'numeric', month: '2-digit', day: '2-digit'
  }
  return new Intl.DateTimeFormat(locale, options ).format(date)
}
```